_search-plugins/ubi/schemas.md

```diff
@@ -0,0 +1,228 @@
+graph LR
+style L fill:none,stroke-dasharray: 5 5
+subgraph L["`*Legend*`"]
+  style ss height:150px
+  subgraph ss["Standard Search"]
+   direction LR
+    
+   style ln1a fill:blue
+   ln1a[ ]--->ln1b[ ];
+  end
+  subgraph ubi-leg["UBI data flow"]
+   direction LR
+   
+   ln2a[ ].->|"`**UBI interaction**`"|ln2b[ ];
+   style ln1c fill:red
+   ln1c[ ]-->|<span style="font-family:Courier New">query_id</span> flow|ln1d[ ];
+  end
+end
+linkStyle 0 stroke-width:2px,stroke:#0A1CCF
+linkStyle 2 stroke-width:2px,stroke:red
+```
+```mermaid
+%%{init: {
+  "flowchart": {"htmlLabels": false},
+
+  } 
+}%%
+graph TB
+
+User--1) <i>raw search string</i>-->Search;  
+Search--2) <i>search string</i>-->Docs 
+style OS stroke-width:2px, stroke:#0A1CCF, fill:#62affb, opacity:.5
+subgraph OS[OpenSearch Cluster fa:fa-database]
+  style E stroke-width:1px,stroke:red
+  E[(&emsp;<b>UBI Events</b>&emsp;)]
+  style Docs stroke-width:1px,stroke:#0A1CCF
+  style Q stroke-width:1px,stroke:red
+  Docs[(Document Index)] -."3) {<i>DSL</i>...} & [<i>object_id's</i>,...]".-> Q[(&emsp;<b>UBI Queries</b>&emsp;)]; 
+  Q -.4) <span style="font-family:Courier New">query_id</span>.-> Docs ;  
+end
+
+Docs -- "5) <i>return</i> both <span style="font-family:Courier New">query_id</span> & [<i>objects</i>,...]" --->Search ;
+Search-.6) <span style="font-family:Courier New">query_id</span>.->U;
+Search --7) [<i>results</i>, ...]--> User
+
+style *client-side* stroke-width:1px, stroke:#D35400
+subgraph "`*client-side*`"
+  style User stroke-width:4px, stroke:#EC636
+  User["`**User**`" fa:fa-user]
+  App
+  Search  
+  U
+  style App fill:#D35400,opacity:.35, stroke:#0A1CCF, stroke-width:2px
+  subgraph App[&emsp;&emsp;&emsp;&emsp;&emsp;&emsp;&emsp;UserApp fa:fa-store]
+    style Search stroke-width:2px, stroke:#0A1CCF
+    Search(&emsp;Search Client&emsp;)    
+    style U stroke-width:1px,stroke:red
+    U(&emsp;<b>UBI Client</b>&emsp;)
+  end
+end
+
+User -.8) <i>selects</i> <span style="font-family:Courier New">object_id:123</span>.->U;
+U-."9) <i>index</i> event:{<span style="font-family:Courier New">query_id, onClick, object_id:123</span>}".->E;
+
+linkStyle 1,2,0,6 stroke-width:2px,fill:none,stroke:#0A1CCF
+linkStyle 3,4,5,8 stroke-width:2px,fill:none,stroke:red
+```
+{% endcomment %}
+
+- The **Search Client** is in charge of searching and then receiving *objects* from a document index in OpenSearch.
+  <br/>(1, 2, **5** and 7, in the preceding diagram)
+
+Step **5** is in bold because it denotes UBI-specific additions like [`query_id`](#query_id) to standard OpenSearch interactions.
+ {: .note}
+- If activated in the `ext.ubi` stanza of the search request, the **User Behavior Insights** plugin manages the **UBI Queries** store in the background, indexing each query, ensuring a unique [`query_id`](#query_id) along with all returned resultant [`object_id`](#object_id)'s, and then passing the `query_id` back to the **Search Client** so that events can be linked to this query.
```

Comment:
```suggestion
- If activated in the `ext.ubi` stanza of the search request, the **User Behavior Insights** plugin manages the **UBI queries** store in the background, indexing each query, ensuring a unique [`query_id`](#query_id) along with all returned resultant [`object_id`](#object_id)'s, and then passing the `query_id` back to the **Search client** so that events can be linked to this query.
```